### PR TITLE
CVE-2017-9793

### PR DESCRIPTION
**Pull request by moiseslorap**
APACHE Struts - REST Plugin Dependency Vulnerability

### CODE REVIEW

File: cves/CVE-2017-9793.yml [hunk 1]
```diff
@@ -139,19 +137,21 @@ discovered:
...
 subsystem:
   question: |
     What subsystems was the mistake in?
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged. Examples: "clipboard", "gpu", "ssl", "speech", "renderer"
-  answer:
-  name:
+  answer: Dependencies XML
+  name: pom.xml
```
Comment (MattThyng) [inline cves/CVE-2017-9793.yml]: I would look for a folder rather than an individual file for the subsystem.

**Comment by moiseslorap** [comment]
It was in the root folder of the source code repository. Forgot to add that
detail.

Thank you,
Moisés.

File: cves/CVE-2017-9793.yml [hunk 2]
```diff
@@ -225,3 +231,9 @@ mistakes:
     every one. Write a thoughtful entry here that those in the software
     engineering industry would find interesting.
   answer:
+    This vulnerability was exploited due to an outdated library dependency that
+    the REST Plugin on this application utilizes. The ideal scenario to avoid this sort
+    of vulnerability is to restrict the use of third party dependencies, this can be mitigated
+    by either redesigning the architecture of the plugin or attempting to design an internal library
+    that was thorougly tested before production. It is always important to be wary of third-party
+    dependencies which are constantly targeted by attackers, especially for open source software like Struts.
```
Comment (MattThyng) [inline cves/CVE-2017-9793.yml:239]: Do you think trust boundaries could have helped mitigate this issue?

**Comment by mll8657** [comment]
I was going to comment on how the description section was quite similar to the CVE description on the official sites and I wished it explained the tools and libraries better, but it seems that the description isn't a part of this pull request. Was your description section already filled out for you? Maybe it needs touching up anyways.

File: cves/CVE-2017-9793.yml [hunk 3]
```diff
@@ -162,11 +162,14 @@ interesting_commits:
     If there are no interesting commits, demonstrate that you completed this section
     by explaining what happened between the VCCs and the fix.
   answer:
+    The commit I highlighted was very important for the fix for this vulnerability as
```
Comment (mll8657) [inline cves/CVE-2017-9793.yml]: I might split this up into two sentences. The sentence felt like it ran on for quite a while and was hard to follow.

**Comment by mll8657** [comment]
+1 upvotes

File: cves/CVE-2017-9793.yml (hunk 2 above)
Comment (moiseslorap) [reply cves/CVE-2017-9793.yml:239]: Yes indeed, if trust boundaries were implemented where the application would validate input that affects critical system functions, and deny anything that is not trustful this type of vulnerability could have been avoided.

File: cves/CVE-2017-9793.yml (hunk 1 above)
Comment (moiseslorap) [reply cves/CVE-2017-9793.yml]: Fix should address that it is at the root.

File: cves/CVE-2017-9793.yml (hunk 3 above)
Comment (moiseslorap) [reply cves/CVE-2017-9793.yml]: Thanks, will edit.

**Comment by moiseslorap** [comment]
> I was going to comment on how the description section was quite similar to the CVE description on the official sites and I wished it explained the tools and libraries better, but it seems that the description isn't a part of this pull request. Was your description section already filled out for you? Maybe it needs touching up anyways.

Yes it was already provided. But I have elaborated.

**Pull request merged**